[cxx-interop] Add support for C++ comparison operators.

zoecarver opened the pull request:
Adds support for C++ operators: `<`, `>`, `==`, `!=`, `<=`, and `>=`.

zoecarver committed:
[cxx-interop] Add support for C++ comparison operators.

Adds support for C++ operators: `<`, `>`, `==`, `!=`, `<=`, and `>=`.

zoecarver commented:
@swift-ci please smoke test

MForster commented on lib/ClangImporter/ImportName.cpp:
```diff
@@ -1427,6 +1427,12 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
+    case clang::OverloadedOperatorKind::OO_EqualEqual:
+    case clang::OverloadedOperatorKind::OO_ExclaimEqual:
+    case clang::OverloadedOperatorKind::OO_LessEqual:
+    case clang::OverloadedOperatorKind::OO_GreaterEqual:
```
Can we keep this list (and also the test cases) in the same order as https://github.com/llvm/llvm-project/blob/master/clang/include/clang/Basic/OperatorKinds.def#L58,L104?

zoecarver replied:
Good idea. I did this as a separate commit.

zoecarver committed:
[cxx-interop] Re-order operators.

Re-order operator case statements and tests. The order now follows the order defined in `llvm-project/clang/include/clang/Basic/OperatorKinds.def`.

Also, adds operator character(s) in parentheses.

zoecarver commented:
@swift-ci please smoke test and merge.

zoecarver commented:
@swift-ci please smoke test and merge.

zoecarver commented:
@swift-ci please smoke test and merge.
